optuna/integration/_allennlp.py

```diff
@@ -0,0 +1,14 @@
+from typing import Any
+from typing import Callable
+
+
+class EpochCallback:
```

Comment:
This is introduced for building documentation.

This PR adds a decorator `@EpochCallback.register` to `AllenNLPPruningCallback`,
which is undefined in document CI as it doesn't install `allennlp`.